src/pages/projects/[pid]/document/[documentId]/vote/[id].js

```diff
@@ -49,66 +45,61 @@ export default function Index({project, document, change, changeVotes, userVoteP
 		return true;
 	}
 
-	const incrementVote = async () => {
-		if (!userCanVote()) {
-			return;
-		}
-		const result = await voteOnChange(change.cid, {vote: 1}, getCookie("stytch_session_jwt"));
-		if (!result) {
-			return;
-		}
-		if (userVote === 0) {
-			setTotalVotes(totalVotes + 1);
-		}
-		setUserVote(1);
-	};
-
-	const decrementVote = async () => {
-		if (!userCanVote()) {
-			return;
-		}
-		const result = await voteOnChange(change.cid, {vote: -1}, getCookie("stytch_session_jwt"));
-		if (!result) {
-			return;
-		}
-		if (userVote === 0) {
-			setTotalVotes(totalVotes + 1);
-		}
-		setUserVote(-1);
-	};
+	const ResultsSection = () => (
+		<>
+			<h2 className="mt-4 text-lg">Results</h2>
+			<AwaitResults/>
+		</>
+	);
 
 	return (
-		<Layout>
-			<div className="flex max-h-screen">
-				<div className="w-1/4 pr-6 mt-10">
-					<h1 className="text-4xl font-extrabold">{change.name}</h1>
-					<div className="col-1 space-y-4 space-x-2 mt-8">
-						<Button variant="outline" onClick={goToVotes}>
-							Back to Changes
-						</Button>
-						<Button onClick={goToDocument}>
-							See Document
-						</Button>
-					</div>
-					<div className="mt-4 ml-2 text-2xl font-bold pt-4">Vote Here</div>
-					<div className="py-2">
-						<Button variant={userVote === -1 ? "" : "outline"} onClick={decrementVote}>
-							-1
-						</Button>
-						<Button variant={userVote === 1 ? "" : "outline"} className="mx-8" onClick={incrementVote}>
-							+1
-						</Button>
-					</div>
-					<div className="text-xl">Total Votes: {totalVotes}</div>
-				</div>
-				<div className="flex-1 max-w-full max-h-screen p-4 ml-2 border-l mt-14 lg:prose-md">
-					<div className="h-full px-5 -mt-6 overflow-x-scroll overflow-y-scroll">
-						{files.map(renderFile)}
-					</div>
-				</div>
+		<NewLayout>
+			<div className="flex flex-col justify-center pb-6 bg-neutral-100">
+				<main
+					className="flex flex-col items-start p-8 w-full bg-white rounded-3xl max-md:px-5 max-md:max-w-full">
+					{!change.closed && !change.merged ? (
+						<VotingOngoingBadge/>
+					) : change.closed ? (
+						<VoteNotPassedBadge/>
+					) : (
+						<VotePassedBadge/>
+					)}
```

Comment:
Should be encapsulated as a `VoteStatusBadge` that decides all 3 states. Removes need for 5 badges.

Reply:
condensed